nikola/utils.py

```diff
@@ -1870,17 +1870,21 @@ def sort_posts(posts, *keys):
 
 
 def smartjoin(join_char: str, string_or_iterable) -> str:
-    """Join string_or_iterable with join_char if it is not a string already.
+    """Join string_or_iterable with join_char if it is iterable; otherwise converts it to string.
 
     >>> smartjoin('; ', 'foo, bar')
     'foo, bar'
     >>> smartjoin('; ', ['foo', 'bar'])
     'foo; bar'
+    >>> smartjoin(' to ', ['count', 42])
+    'count to 42'
     """
     if isinstance(string_or_iterable, (unicode_str, bytes_str)):
         return string_or_iterable
+    elif hasattr(string_or_iterable, '__iter__'):
```

Comment:
`isinstance(string_or_iterable, collections.Iterable)` would look nicer IMO

Reply:
Is this actually equivalent?

Reply:
yes

Reply:
Good to know. I changed it.